### ISSUE TITLE
False positive: UnconditionalJumpStatementInLoop with Elvis operator

### ISSUE THREAD

**Issue by MiSikora** [detekt/detekt#4134]
## Expected Behavior

The Elvis operator in a conditional code like this should be accepted:

```kotlin
fun function(items: List<Int?>): String? {
    for (item in items) {
        val intItem = item ?: continue
        return intItem.toString()
    }
    return null
}
```

## Observed Behavior

It is reported as `UnconditionalJumpStatementInLoop`.

## Context

Original code is a little bit more complex where there is data processing between `return` and `continue`. It can be generally avoided using sequences or the `let` function.

## Your Environment

* Version of detekt used: `1.18.1`
* Version of Gradle used (if applicable): `7.2`
* Operating System and version: `macOS 11`

**Comment by MiSikora** [comment]
I'll gladly make a PR for this sometime soon.

**Comment by BraisGabin** [comment]
> I'll gladly make a PR for this sometime soon.

That would be great :)
